analysis/interface/topAnalysis.h

```diff
@@ -8,26 +8,56 @@
 class topAnalysis : public nanoAnalysis
 {
 protected:
+  //Histogram
+  TH1D* h_nevents;
+  TH1D* h_genweights;
+  TH1D* h_weights;
+  TH1D* h_cutFlow;
+
+  //Variables
+  TLorentzVector b_lep1, b_lep2, b_dilep, b_jet1, b_jet2;
+  TParticle recolep1, recolep2;
+  int b_lep1_pid, b_lep2_pid;
+  float b_jet1_CSVInclV2, b_jet2_CSVInclV2;
+
   std::vector<Float_t> b_csvweights;
-  float b_btagweight;
-  //TParticle GetTParticle(int pdgId, int idx);
+
+  int b_nvertex, b_step, b_channel, b_njet, b_nbjet;
+  bool b_step1, b_step2, b_step3, b_step4, b_step5, b_step6, b_step7;  
+  float b_met, b_weight, b_genweight, b_puweight, b_btagweight;
+  float b_mueffweight, b_mueffweight_up, b_mueffweight_dn,
+        b_eleffweight, b_eleffweight_up, b_eleffweight_dn;
+  float b_tri, b_tri_up, b_tri_dn;
+
+  //Triggers
+  Bool_t b_trig_m, b_trig_m2,  b_trig_e, b_trig_mm, b_trig_em, b_trig_ee;
+
+  //Tools
+//  TH1D* hist_mc;
+  MuonScaleFactorEvaluator muonSF_;
+  ElecScaleFactorEvaluator elecSF_;
+
+  //For C meson
+  std::vector<TLorentzVector> d0s;
+  TLorentzVector b_d0;
+  std::vector<float> b_d0_lepSV_lowM;
+  std::vector<float> b_d0_lepSV_correctM;
+  std::vector<float> b_d0_lepSV_dRM;
 
 public:
+  Bool_t m_isDL, m_isSL_e, m_isSL_m;
+  Bool_t m_isHadAna;
+
   enum TTLLChannel { CH_NOLL = 0, CH_MUEL, CH_ELEL, CH_MUMU };
 
-  //Bool_t lumiCheck();
+  bool EventSelection(Bool_t m_isHadAna = false);
```

Comment:
Instead of returning a bool, why not an int for the step passed, then you don't need to send it m_isHadAna, you can just check that you got to the step you're interested in (also m_* should only be used for member variables, ie variables belonging to the class).